[Notification] module readme (#51657)
metabase/metabase: src/metabase/notification/README.md

3. **Subscriptions**: One or more triggers that determine when the notification should be sent

Here is a diagram of the basic flow:
```

     +-------------------+
     |   Subscription    |
     +-------------------+
               |
               | Notification info
               |
+==============|================+
| NOTIFICATION |                |
|              v                |
|      +------------------+     |
|      |     Execute      |     |
|      +------------------+     |
|                               |
+===============================+
               |
               | Notification payload
               |
+==============|=================+
| CHANNEL      |                 |
|              |                 |
|              v                 |
|      +------------------+      |
|      |    Template      |      |
|      |     Engine       |      |
|      +------------------+      |
|              |                 |
|              | Message         |
|              v                 |
|      +------------------+      |
|      |   Destination    |      |
|      +------------------+      |
|                                |
+================================+
```

> You can play with notifications with this simpple code

Reviewer: ```suggestion
You can play with notifications with this simple code
```

```clojure
(require '[metabase.notification.test-util :as notification.tu])
(require '[metabase.notification.core :as notification])

(notification.tu/with-card-notification
  [notification {:card          {:dataset_query (mt/mbql-query users)}
                 :subscriptions [{:type          :notification-subscription/cron
                                  :cron_schedule "0 0 0 * * ?"}]
                 :handlers      [{:channel_type :channel/slack
                                  :recipients   [{:type     :notification-recipient/raw-value
                                                  :details  {:value "#general"}}]}
                                 {:channel_type :channel/email
                                  :recipients   [{:type    :notification-recipient/user
                                                  :user_id (mt/user->id :crowberto)}]}]}]
  (notification/send-notification! notification :notification/sync? true))
```

> This creates a card notification with 1 cron subscription and 2 handlers and send it synchronously.

Reviewer: ```suggestion
This creates a card notification with 1 cron subscription and 2 handlers and sends it synchronously.
```

## Subscription

A subscription is a trigger that determines when the notification should be sent.
Currently we have 2 types of subscriptions:
- `:notification-subscription/cron`: a cron schedule that determines when the notification should be sent
    - Cron subscriptions are managed by quartz scheduler. See `metabase.task.notification` for more details
- `:notification-subscription/system-event`: a system event that determines when the notification should be sent

## Payload

> Once a notification is triggered, we need to execute the payload to get the data needed to render the notification.

Reviewer: Execute seems like the wrong verb here, execution comes later?

```suggestion
Once a notification is triggered, we need to construct a payload containing the data needed to render the notification.
```
